nodejs/npmmirror.lua

```diff
@@ -0,0 +1,145 @@
+    author = "yimiaoxiehou",
+    version = "0.0.1",
+    description = "install Node.js use https://cdn.npmmirror.com",
+    updateUrl = "https://raw.githubusercontent.com/version-fox/version-fox-plugins/main/node/npmmirror.lua",
+}
+
+function PLUGIN:PreInstall(ctx)
+    local version = ctx.version
+
+    if version == "latest" then
+        local lists = self:Available({})
+        version = lists[1].version
+    end
+    local arch_type = ARCH_TYPE
+    local ext = ".tar.gz"
+    local osType = OS_TYPE
+    if arch_type == "amd64" then
+        arch_type = "x64"
+    end
+    if OS_TYPE == "windows" then
+        ext = ".zip"
+        osType = "win"
+    end
+    local filename = FileName:format(version, osType, arch_type, ext)
+    local baseUrl = NodeBaseUrl:format(version)
+
+    local resp, err = http.get({
+        url = baseUrl .. "SHASUMS256.txt"
+    })
+    if err ~= nil or resp.status_code ~= 200 then
+        error("get checksum failed")
+    end
+    local checksum = get_checksum(resp.body, filename)
+    return {
+        version = version,
+        url = baseUrl .. filename,
+        sha256 = checksum,
+    }
+end
+
+function compare_versions(v1o, v2o)
+    local v1 = v1o.version
+    local v2 = v2o.version
+    local v1_parts = {}
+    for part in string.gmatch(v1, "[^.]+") do
+        table.insert(v1_parts, tonumber(part))
+    end
+
+    local v2_parts = {}
+    for part in string.gmatch(v2, "[^.]+") do
+        table.insert(v2_parts, tonumber(part))
+    end
+
+    for i = 1, math.max(#v1_parts, #v2_parts) do
+        local v1_part = v1_parts[i] or 0
+        local v2_part = v2_parts[i] or 0
+        if v1_part > v2_part then
+            return true
+        elseif v1_part < v2_part then
+            return false
+        end
+    end
+
+    return false
+end
+
+function get_checksum(file_content, file_name)
+    for line in string.gmatch(file_content, '([^\n]*)\n?') do
+        local checksum, name = string.match(line, '(%w+)%s+(%S+)')
+        if name == file_name then
+            return checksum
+        end
+    end
+    return nil
+end
+
+function PLUGIN:Available(ctx)
+    local resp, err = http.get({
+        url = VersionSourceUrl
+    })
+    if err ~= nil or resp.status_code ~= 200 then
+        return {}
+    end
+    local body = json.decode(resp.body)
+    local result = {}
+    for _, v in ipairs(body) do
+        table.insert(result, {
+            version = string.gsub(v.version, "^v", ""),
+            note = v.lts and "LTS" or "",
+            addition = {
+                {
+                    name = "npm",
+                    version = v.npm,
+                }
+            }
+        })
+    end
+    table.sort(result, compare_versions)
+    return result
+end
+
+--- Expansion point
+function PLUGIN:PostInstall(ctx)
+    local rootPath = ctx.rootPath
+    local sdkInfo = ctx.sdkInfo['nodejs']
+    local path = sdkInfo.path
+    local version = sdkInfo.version
+    local name = sdkInfo.name
```

Comment:
You can delete these lines safely, I just wrote them to test the extension, they don't make sense.